shelley/chain-and-ledger/formal-spec/multi-sig.tex

```diff
@@ -738,59 +802,62 @@ \subsection{Native Script Interpreter}
-for such a native script implementation. The type $\ScriptMSig$ is defined as a
+script as a data type that can then be interpreted
+directly. Figure~\ref{fig:types-msig} shows the types and functions
+that are needed for such a native script implementation. The type $\ScriptMSig$ is defined as a
 tree-structure which is either a single signature leaf node or a list of values
-of type $\ScriptMSig$, which either requires all signatures to validate, at
-least one of the signatures to validate or at least the threshold value of $m$
-signatures to validate.
+of type $\ScriptMSig$.  This either requires all signatures to be validated, at
+least one of the signatures to be validated, or at least the threshold value of $m$
+signatures to be validated.
 
 \subsection{Lower Level Implementation Details}
 \label{sec:lower-level-impl}
 
 For each new type of witness, there will be a requirement to represent such a
 witness on-chain and allow for identification when deserializing such a
-witness. For this, there will be a language specific tag for each witness (and
+witness. For this, there will be a language-specific tag for each witness (and
 staking reference) in its serialized form. This tag will also be part of the
 hash to allow for easy identification of the language type.
 
-As an example, one could tag key hash payment object and staking reference with
-a tag $0$, for native multi-sig with tag $1$ and simple Plutus scripts with tag
-$2$. Every new language would incur using a new tag, changes to the details of
-how payment or staking references are treated \emph{within} an existing language
-framework will be changed via a software update, not an additional tag.
+As an example, one could tag key hash payment object and staking references with
+the tag $0$, native multi-signature scripts with the tag $1$, and simple Plutus scripts with the tag
+$2$. Every new language would require a new tag. Changes to the payment or staking reference details
+that are treated \emph{within} an existing language
+framework would be changed via a software update, rather than an additional tag.
 
 \section{Summary}
 \label{sec:summary}
 
-The presented script based multi-signature scheme does not require the scripts
-to use any cryptographic primitives. It requires only the ability to compare the
-required keys to those that actually signed the transaction. The worst case
+The script-based multi-signature scheme that is presented here does not require the scripts
+to use any cryptographic primitives. Rather, it requires only the ability to compare the
+required keys to those that actually signed the transaction. The worst-case
 potential calculation cost can therefore be calculated statically in advance and
-can be added to the transaction fees or as gas cost. The scripts can be realized
-with only limited requirements on functionality on the script language.
-
+can be added to the transaction fees or as gas cost. \khcomment{Assumes transactions are independent.}
```

Comment:
I don't understand this. The creator of the transaction knows how many signatures are present and knows the script, which means that the creator can statically check for how long the script can run (actually for `MSig`, how many reduction steps are required for False / True). How is this influenced by other transactions ?

Reply:
If there was a dependency between transactions.  It may be that by design this can't happen.